packages/test/test-end-to-end-tests/src/test/SummarizerWithDataStoreRealization.spec.ts

```diff
@@ -0,0 +1,278 @@
+    return mixinSummaryHandler(async (runtime: FluidDataStoreRuntime) => {
+        const obj: Provider<SearchContent> = (await DataObject.getDataObject(runtime)) as Provider<SearchContent>;
+        const searchObj = obj.SearchContent;
+        if (searchObj === undefined) {
+            return undefined;
+        }
+        // ODSP parser requires every search blob end with a line-feed character.
+        const searchContent = await searchObj.getSearchContent();
+        if (searchContent === undefined) {
+            return undefined;
+        }
+        const content = searchContent.endsWith("\n") ? searchContent : `${searchContent}\n`;
+        return {
+            // This is the path in snapshot that ODSP expects search blob (in plain text) to be for components
+            // that want to provide search content.
+            path: ["_search", "01"],
+            content,
+        };
+    }, factory);
+}
+
+/**
+ * Validates whether or not a GC Tree Summary Handle should be written to the summary.
+ */
+describeNoCompat("Summary with Search Blobs and DataStore realization during Summarization",
+ (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject1;
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory);
+    };
+
+    async function waitForSummary(summarizer: ISummarizer): Promise<string> {
+        // Wait for all pending ops to be processed by all clients.
+        await provider.ensureSynchronized();
+        const summaryResult = await summarizeNow(summarizer);
+        return summaryResult.summaryVersion;
+    }
+
+    describe("Realize DataStore during Search and make sure pendingSummaries does not get corrupted", () => {
+        beforeEach(async () => {
+            provider = getTestObjectProvider({ syncSummarizer: true });
+            mainContainer = await createContainer();
+            // Set an initial key. The Container is in read-only mode so the first op it sends will get nack'd and is
+            // re-sent. Do it here so that the extra events don't mess with rest of the test.
+            mainDataStore = await requestFluidObject<TestDataObject1>(mainContainer, "default");
+            mainDataStore._root.set("anytest", "anyvalue");
+            await waitForContainerConnection(mainContainer);
+        });
+
+        it("No Summary Upload Error when DS gets realized between summarize and completeSummary", async () => {
+            const summarizerClient = await createSummarizer(provider, mainContainer);
+            await provider.ensureSynchronized();
+            mainDataStore.matrix.setCell(0, 0, "value");
+
+            // Here are the steps that would cause bug 1633 to repro:
+            // Additional information: https://dev.azure.com/fluidframework/internal/_workitems/edit/1633
```

Comment:
I don't think we should add ADO links since those are internal to MS.

Reply:
I thought it could be linked...